solutions/usaco-623.mdx

```diff
@@ -0,0 +1,115 @@
+---
+id: usaco-623
+source: USACO Gold 2016 February
+title: Fenced In
+author: Jeffrey Zhang
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_fencedin_gold_feb16.html)
+
+## Solution
+
+The $n$ vertical and $m$ horizontal fences divide the large rectangular pasture into $(n+1)(m+1)$ grids. To help us better comprehend the problem, it is easier for us to visualized the input in terms of graphs: the grids themselves are the nodes, and the fence lengths of each 4 surrounding sides are the edge lengths that can be constructed to their neighboring nodes. For a cow to travel from any node to another, all we need is to construct a tree graph. And to ensure the edges picked are at a minimum, we just need to construct a minimum spanning tree, which can be solved by the [Kruskal's MST algorithm](https://usaco.guide/gold/mst#kruskals).
```

Comment:
this seems to just repeat the official?

Reply:
The problem is in the MST module, and I don't think there are other ways to solve this...

Reply:
if that's the case, then just add an implementation- no need for a solution